src/IndexedArrays.jl

```diff
@@ -85,6 +85,23 @@ end
 
 copy{T}(ia::IndexedArray{T}) = IndexedArray{T}(copy(ia.items))
 
-export AbstractIndexedArray, IndexedArray, IndexedArrayError, findfirst!
+"`swap!(ia::IndexedArray, to::Int, from::Int)` interchange/swap the values on the indices `to` and `from` in the `IndexedArray`"
+function swap!(ia::IndexedArray, to::Int, from::Int)
+    if to == from && checkbounds(ia,to)
```

Comment:
Actually, I would have put the `checkbounds` as the first line inside the `if` block, not as part of the conditional, which is how it is typically done.  It is interesting that the current code also works correctly though, which has led me to filing https://github.com/JuliaLang/julia/issues/13334